test/server/cards/01_SOR/units/InfernoFourUnforgetting.spec.ts

```diff
@@ -0,0 +1,57 @@
+describe('Inferno Four - Unforgetting', function() {
+    integration(function(contextRef) {
+        describe('Inferno Four - Unforgetting\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['inferno-four#unforgetting'],
+                        deck: ['foundling', 'pyke-sentinel', 'atst', 'cartel-spacer', 'wampa'],
+                    },
+                    player2: {
+                        spaceArena: ['tie-advanced'],
+                    }
+                });
+            });
+
+            it('While playing/defeating lets you look at the top 2 cards of the deck and decide whether to put either them on the bottom or top of deck in any order.', function () {
+                const { context } = contextRef;
+                let preSwapDeck = context.player1.deck;
+
+                // Case 1 on play move top card to bottom
+                context.player1.clickCard(context.infernoFour);
+                context.player1.clickPrompt('pyke-sentinel-bottom');
```

Comment:
At each stage we should use `expect().toHaveExactPromptButtons()` to confirm the set of available prompt buttons. Also, at least for this test we should confirm that cards haven't moved until all selections are done.